controllers/propagator/propagation.go

```diff
@@ -901,6 +937,83 @@ func (r *PolicyReconciler) processTemplates(
 			ManagedClusterName: decision.ClusterName,
 		}
 
+		// Handle value encryption initialization
+		if encryptionEnabled {
+			usesEncryption := templates.UsesEncryption(
+				policyT.ObjectDefinition.Raw, templateCfg.StartDelim, templateCfg.StopDelim,
+			)
+			// Initialize AES Key and initialization vector
+			if usesEncryption && !encryptionInitialized {
+				log.V(1).Info("Found an object definition requiring encryption. Handling encryption keys.")
+				// Get/generate the encryption key
+				encryptionKey, err := r.getEncryptionKey(decision.ClusterName)
+				if err != nil {
```

Comment:
I am thinking we want to return the error here rather than keep going on since if the policy uses encryption but the Propagator can't get the key to encrypt the data, then there's no way to proceed. `handleDecisionWrapper` will retry the `handleDecision` method anyways.